tfworker/definitions.py

```
@@ -0,0 +1,224 @@
+        body,
+        deployment,
+        global_remote_vars,
+        global_template_vars,
+        global_terraform_vars,
+        providers,
+        repository_path,
+        temp_dir,
+        limited=False,
+    ):
+        self.tag = definition
+        self._body = body
+        self._path = body.get("path")
+        self._remote_vars = self.make_vars(
+            body.get("remote_vars", collections.OrderedDict()), global_remote_vars
+        )
+        self._template_vars = self.make_vars(
+            body.get("template_vars", collections.OrderedDict()), global_template_vars
+        )
+        self._terraform_vars = self.make_vars(
+            body.get("terraform_vars", collections.OrderedDict()), global_terraform_vars
+        )
+
+        self._deployment = deployment
+        self._repository_path = repository_path
+        self._providers = providers
+        self._temp_dir = temp_dir
+        self._limited = limited
+
+    @property
+    def body(self):
+        return self._body
+
+    @property
+    def limited(self):
+        return self._limited
+
+    @property
+    def path(self):
+        return self._path
+
+    def prep(self, backend):
+        """ prepare the definitions for running """
+        repo = Path(f"{self._repository_path}/{self.path}".replace("//", "/"))
+        target = Path(f"{self._temp_dir}/definitions/{self.tag}".replace("//", "/"))
+        target.mkdir(parents=True, exist_ok=True)
+
+        if not repo.exists():
+            click.secho(
+                f"Error preparing definition {self.tag}, path {repo.resolve()} does not"
+                " exist",
+                fg="red",
+            )
+            sys.exit(1)
+
+        # Prepare variables
+        self._template_vars["deployment"] = self._deployment
+        self._terraform_vars["deployment"] = self._deployment
+
+        # Put terraform files in place
+        for tf in repo.glob("*.tf"):
+            shutil.copy("{}".format(str(tf)), str(target))
```

we should raise an error here if terraform.tf or worker-locals.tf exists as it will cause a problem as worker expects to be able to generate these files, I'm okay deferring this to a future story since it's a scope change

---

I think I got this 👍